Server.Tests/ParserTests.cs

```diff
@@ -89,5 +89,70 @@ public void Request_With_Body_Should_Return_Correct_Body()
             Assert.AreEqual("1.1", actual.Version);
             Assert.AreEqual("THIS IS THE BODY", actual.Body);
         }
+
+
+        [TestMethod]
+        public void Requests_With_Specific_Method_Should_Return_Correct_Method_Value_Test()
+        {
+            // 1. Arrange
+            RequestParser parser = new RequestParser();
+            var GETrequestString = @"GET /one/two?three=4 HTTP/1.1";
+            var POSTrequestString = @"POST /one/two?three=4 HTTP/1.1";
+            var PUTrequestString = @"PUT /one/two?three=4 HTTP/1.1";
+            var DELETErequestString = @"DELETE /one/two?three=4 HTTP/1.1";
+            var OPTIONSrequestString = @"OPTIONS /one/two?three=4 HTTP/1.1";
+            var HEADrequestString = @"HEAD /one/two?three=4 HTTP/1.1";
+
+            // 2. Act
+            var actualGet = parser.Parse(GETrequestString);
+            var actualPost = parser.Parse(POSTrequestString);
+            var actualPut = parser.Parse(PUTrequestString);
+            var actualDelete = parser.Parse(DELETErequestString);
+            var actualOptions = parser.Parse(OPTIONSrequestString);
+            var actualHead = parser.Parse(HEADrequestString);
+
+            // 3. Assert
+            Assert.AreEqual(Method.Get, actualGet.Method);
+            Assert.AreEqual(Method.Post, actualPost.Method);
+            Assert.AreEqual(Method.Put, actualPut.Method);
+            Assert.AreEqual(Method.Delete, actualDelete.Method);
+            Assert.AreEqual(Method.Options, actualOptions.Method);
+            Assert.AreEqual(Method.Head, actualHead.Method);
+        }
+
+        [TestMethod]
+        public void Request_With_Specific_Header_Should_Return_Correct_Header_Value_Test()
+        {
+            // 1. Arrange
+            RequestParser parser = new RequestParser();
```

Comment:
This one is better, but still borderline. We have one action method, and then we do multiple asserts on the result.